### PR TITLE
feat: lookup project tsconfig

### LINKED ISSUE #2062: Clarify documentation on tsconfig usage
Per https://github.com/vega/ts-json-schema-generator/pull/168 I understand that `tsconfig.json` is not read by default. This was confusing to me, as I have a project where building schemas failed due to this error:

```
➜ npx ts-json-schema-generator --path src/types/config.ts --type Config
error TSJ - 108: Type check error

  /Users/werner/Documents/Projects/test/node_modules/@types/selenium-webdriver/bidi/captureScreenshotParameters.d.ts:9:5
    9     #map: Map<string, any>;
          ~~~~
    Private identifiers are only available when targeting ECMAScript 2015 and higher.
```

Yet in my `tsconfig.json` I have specified ES2023 as a target.

I only then realized that I needed to explicitly pass the `tsconfig.json` file to the command to make it work. Just from the CLI args I would have assumed that "Custom tsconfig.json path" means: "Any config other than `tsconfig.json`".

Is that behavior intentional? (One would assume that proper type compilation also requires knowing about the project's build target, but maybe this is not as big of a problem as I think it is for most users.)

It would be great if the documentation could be updated to reflect that aspect. Maybe I'm the first one to get bitten by this, but from a quick review I haven't found similar issues. Let me know how you think this could be best addressed and I can provide a PR!

### PR DESCRIPTION
**Pull request by slhck**
This looks up the default tsconfig.json file for the current project when used from the CLI, to avoid possible compliation errors. Fixes #2062

### CODE REVIEW

**Review by arthurfiorette — APPROVED** [review]

**Comment by arthurfiorette** [comment]
please create a test for when no tsconfig files can be found

**Comment by slhck** [comment]
If no tsconfig.json file is found, the behavior is exactly the same as before this commit — you can see the only change is the assignment in the arguments. Since you have no real CLI tests suite, I'm not sure how this could be easily integrated.

**Comment by slhck** [comment]
(Or do you mean just in the unit test, because of the missing code coverage for the "not found" case?)

**Comment by slhck** [comment]
Please check if the new tests suffice.

File: src/Utils/findTsConfig.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Find the tsconfig.json file in the current working directory or any parent directory.
+ * This is the behavior from tsc: https://www.typescriptlang.org/docs/handbook/tsconfig-json.html
+ *
+ * @param directory - The directory to start searching from. Defaults to the current working directory.
+ * @param filename - The filename of the tsconfig.json file. Defaults to "tsconfig.json".
+ * @returns The path to the tsconfig.json file or undefined if no such file was found.
+ */
+export function findTsConfig(directory: string = process.cwd(), filename = "tsconfig.json"): string | undefined {
```
Comment (domoritz) [inline src/Utils/findTsConfig.ts]: Is there any chance that we can import this logic from tsc rather than implementing it? If not, then this is good.

**Review by domoritz — APPROVED** [review]

**Comment by arthurfiorette** [comment]
Maybe (to offload this logic from this codebase) https://www.npmjs.com/package/get-tsconfig?activeTab=readme could be a good idea.

**Comment by domoritz** [comment]
Oh look at the bottom of the read me. They show how to use tsc.

**Comment by slhck** [comment]
Since you don't depend on `tsc`, I don't think that is a viable option, unless you assume users will always have `tsc` present in their modules.

The third-party package you linked to looks like another option (there's a package for everything in Node these days, huh), but it means having another depdendency (with its own dependencies) for something rather trivial, and you don't need its other parsing features. In essence, it does [exactly the same](https://github.com/privatenumber/get-tsconfig/blob/master/src/utils/find-up.ts) as the function I committed.

Feel free to merge or just replace the one function call with the third-party dependency — I am not passionate about this.

**Comment by domoritz** [comment]
We depend on typescript. 

And yes I agree, no to another dep.

**Comment by slhck** [comment]
> We depend on typescript.

Where?

https://github.com/vega/ts-json-schema-generator/blob/46f638ea799cde3d44b113a95bb307c64f2f6021/package.json#L58-L95

**Comment by slhck** [comment]
Oh, never mind, I was looking in the wrong place. Should I update the PR or could you make that two line change yourself?

**Comment by domoritz** [comment]
Go ahead!

**Comment by arthurfiorette** [comment]
> Oh look at the bottom of the read me. They show how to use tsc.

How tf i have never seen this 🤡

**Comment by domoritz** [comment]
Can you refactor the last bit @slhck?

**Comment by slhck** [comment]
Please see the new commit.

**Review by arthurfiorette — APPROVED** [review]

**Comment by swift502** [comment]
I got burned on this last week, happy to see it getting improved. 👏

**Comment by domoritz** [comment]
Thank you!

**Review by domoritz — APPROVED** [review]

**Pull request merged**